optaplanner-persistence/optaplanner-persistence-jackson/src/main/java/org/optaplanner/persistence/jackson/api/score/constraint/ConstraintMatchJacksonJsonDeserializer.java

```diff
@@ -0,0 +1,34 @@
+package org.optaplanner.persistence.jackson.api.score.constraint;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import org.optaplanner.core.api.score.Score;
+import org.optaplanner.core.api.score.constraint.ConstraintMatch;
+import org.optaplanner.persistence.jackson.api.score.DeserializationUtils;
+
+public class ConstraintMatchJacksonJsonDeserializer extends JsonDeserializer<ConstraintMatch> {
+
+    @Override
+    public ConstraintMatch deserialize(JsonParser parser, DeserializationContext context) throws IOException {
+        JsonNode tree = parser.getCodec().readTree(parser);
```

Comment:
readTree() is a DOM approach and it has 2 big disadvantages:
- it parses and buffers the entire subtree. This increases memory usage and lowers performance of the parsing
- the parser can't simply be given to the score deserializer (instead, a new parser is created on the subtree).

I am using the nextToken() approach to avoid this DOM approach.

Christopher, just mentioning this FYI,  to understand that in any type of parser (xml, json, ...) there are often 2 approaches: DOM (= user friendly) and non-DOM (= cpu & memory-friendly).